File: 리트코드/0371-sum-of-two-integers/0371-sum-of-two-integers.java
class Solution {
    public int getSum(int a, int b) {
        String binA = String.format("%32s", Integer.toBinaryString(a)).replace(' ','0');
        String binB = String.format("%32s", Integer.toBinaryString(b)).replace(' ','0');
        
        List<Character> result = new ArrayList<>();

        int carry = 0;

        int sum;

        for (int i = 0; i < 32; i++) {
            // int A = Character.getNumericValue(binA.charAt(31-i));
            // int B = Character.getNumericValue(binB.charAt(31-i));
            int A = binA.charAt(31-i) - '0';
            int B = binB.charAt(31-i) - '0';
            int q1 = A & B;
            int q2 = A ^ B;
            int q3 = q2 & carry;
            sum = carry ^ q2;
            carry = q1 | q3;

            result.add(0,Character.forDigit(sum,2));
        }

        return Integer.parseUnsignedInt(
            result.stream()
                    .map(String::valueOf)
                    .collect(Collectors.joining(""))
        ,2);
    }
}

File: 리트코드/0371-sum-of-two-integers/0371-sum-of-two-integers.kt
class Solution {
    fun getSum(a: Int, b: Int): Int {
        
        val binA:String = String.format("%32s", Integer.toBinaryString(a)).replace(' ', '0');
        val binB:String = String.format("%32s", Integer.toBinaryString(b)).replace(' ', '0');

        var carry = 0
        var sum = 0

        val result: MutableList<Char> = mutableListOf()

        for(i in 0 until 32){
            val A = Character.getNumericValue(binA[31-i])
            val B = Character.getNumericValue(binB[31-i])

            val q1 = A and B
            val q2 = A xor B
            val q3 = q2 and carry
            sum = q2 xor carry
            carry = q1 or q3

            result.add(0, Character.forDigit(sum,2))

        }


        return Integer.parseUnsignedInt(String(result.toCharArray()), 2)
    }
}